core/src/main/java/org/infinispan/util/concurrent/CommandAckCollector.java

```diff
@@ -385,6 +392,7 @@ protected void markReady() {
 
    private class BiasedKeyCollector<T> extends SingleKeyCollector<T> {
       private final Address primaryOwner;
+      @SuppressWarnings("MismatchedQueryAndUpdateOfCollection")
```

Comment:
Add a comment that `unsolicitedAcks.removeIf(backupOwners::remove)` does the query (since it iterates through the collection), this is a bug in IntelliJ.